core/shared/src/main/scala/scalaz/zio/delay/Delay.scala

```diff
@@ -0,0 +1,81 @@
+package scalaz.zio.delay
+
+import java.util.concurrent.TimeUnit
+
+import scalaz.zio.ZIO
+import scalaz.zio.clock.{ Clock, _ }
+import scalaz.zio.delay.Delay._
+import scalaz.zio.duration.Duration
+
+/**
+ * Delay represents how much time an effect must wait before start.
+ * Delays can be relative to current time (a given duration + current time) or an absolute point in time
+ * Delays make waiting less cpu intensive and avoids clock drifts caused by context switching or hibernation
+ */
+sealed trait Delay extends Serializable with Product { self =>
+
+  /**
+   * Computes delay related to current time in milliseconds
+   */
+  final def run: ZIO[Any, Nothing, Duration] = currentTime(unit = TimeUnit.MILLISECONDS).provide(Clock.Live).flatMap {
```

Comment:
This `run` should return `ZIO[Clock, Nothing, Duration]`, and get the current time from the clock (not from `Clock.Live`), or else it won't be testable.